modules/commons/src/test/java/org/apache/synapse/commons/json/JsonValueTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.synapse.commons.json;
+
+import junit.framework.TestCase;
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class JsonValueTest extends TestCase {
+    private static Log logger = LogFactory.getLog(JsonValueTest.class.getName());
+
+    private static final String stringVal = "\"Hello world!\"";
+    private static final String nullVal = "null";
+    private static final String trueVal = "true";
+    private static final String falseVal = "false";
+    private static final String numberVal1 = "12.5";
+    private static final String numberVal2 = "-12.5";
+    private static final String numberVal3 = "123";
+    private static final String failTestVal1 = "123abc";
+    private static final String failTestVal2 = "task";
+    private static final String failTestVal3 = "fun";
+    private static final String failTestVal4 = "abc";
+    private static final String failTestVal5 = "-abc";
+    private static final String failTestVal6 = "\"Hello world!";
+
+    public void testCase1() {
```

Comment:
Shall we add doc comment, explaining what the tests are about?

Reply:
Fixed in #759